### PR TITLE
fix(manager/pub): prioritise SDK version constraint from `pubspec.yaml` over `pubspec.lock`

### PR DESCRIPTION
**Pull request by zeshuaro**
## Changes

For the `pub` manager artifacts update, prioritise to use the SDK version from `pubspec.yaml` over `pubspec.lock` as the tool version constraint to update the artifacts.

## Context

If you define a specific SDK version, e.g. `flutter: 3.16.9` in [pubspec.yaml](https://github.com/zeshuaro/appainter/blob/main/pubspec.yaml#L9), the resulted [pubspec.lock](https://github.com/zeshuaro/appainter/blob/main/pubspec.lock#L1420) would still generate a ranged version `flutter: ">=3.16.9"`.

Right now, Renovate uses the version from the lock file to determine the tool/SDK version constraint, which might not be accurate to what's defined in `pubspec.yaml`. And causing the wrong version to be used to update the artifacts. For example, the current latest Flutter version is 3.19.0 which would conflict to what's defined in `pubspec.yaml` and may fail to update artifacts.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pub/artifacts.spec.ts
Comment (HonkingGoose) [inline lib/modules/manager/pub/artifacts.spec.ts:296]: Can you capitalize the abbreviation `SDK` in the PR title?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by HonkingGoose — COMMENTED** [review]
I'm happy with the PR title now. I'll let the maintainers review the code.

**Review by viceice — APPROVED** [review]

**Pull request merged**
